include/cache.h

```diff
@@ -587,6 +587,31 @@ class Cache : public CassandraStore::Store
   {
     return new DissociateImplicitRegistrationSetFromImpi(impus, impis, timestamp);
   }
+
+  /// Get all the IMPUs for which Homestead has data in its cache.
+  ///
+  /// - When an HSS is in use, this lists all subscribers for which homestead is
+  /// storing data learned from the HSS (which in practise is all subscribers
+  /// that are assigned to this S-CSCF in the HSS).
+  /// - When subscribers are locally provisioned this returns all provisioned
+  /// subscribers.
+  class ListImpus : public CassandraStore::Operation
+  {
+  public:
+    ListImpus() {}
+    virtual ~ListImpus() {}
+
+    virtual std::vector<std::string>& get_impus_ref() { return _impus; }
```

Comment:
It's to make it clear that this function returns a reference to the impus vector (rather than returning it by value and doing a lot of unnecessary copying). I'll spell it out.